mulighetsrommet-api/src/test/kotlin/no/nav/mulighetsrommet/api/services/KafkaSyncServiceTest.kt

```diff
@@ -0,0 +1,119 @@
+            enhet = "2990",
+            avslutningsstatus = avslutningsstatus
+        )
+    }
+
+    fun TiltaksgjennomforingDbo.toDto(): TiltaksgjennomforingDto {
+        return TiltaksgjennomforingDto(
+            id = id,
+            tiltakstype = TiltaksgjennomforingDto.Tiltakstype(
+                id = tiltakstype.id,
+                navn = tiltakstype.navn,
+                arenaKode = tiltakstype.tiltakskode,
+            ),
+            navn = navn,
+            virksomhetsnummer = virksomhetsnummer,
+            startDato = startDato,
+            sluttDato = sluttDato,
+            status = Tiltaksgjennomforingsstatus.fromDbo(today, startDato, sluttDato, avslutningsstatus)
+        )
+    }
+
+    context("oppdater statuser på tiltaksgjennomføringer") {
+        val tiltakstypeRepository = TiltakstypeRepository(database.db)
+        val tiltaksgjennomforingRepository = TiltaksgjennomforingRepository(database.db)
+        val tiltaksgjennomforingKafkaProducer = mockk<TiltaksgjennomforingKafkaProducer>(relaxed = true)
+        val kafkaSyncService =
+            KafkaSyncService(TiltaksgjennomforingRepository(database.db), tiltaksgjennomforingKafkaProducer)
+
+        val startdatoInnenforMenAvsluttetStatus = createTiltaksgjennomforing()
+        val startdatoInnenfor =
+            createTiltaksgjennomforing(avslutningsstatus = Avslutningsstatus.IKKE_AVSLUTTET)
+        val sluttdatoInnenforMenAvbruttStatus = createTiltaksgjennomforing(
+            startDato = lastSuccessDate,
+            sluttDato = lastSuccessDate,
+            avslutningsstatus = Avslutningsstatus.AVBRUTT
+        )
+        val sluttdatoInnenfor = createTiltaksgjennomforing(
+            startDato = lastSuccessDate,
+            sluttDato = lastSuccessDate,
+            avslutningsstatus = Avslutningsstatus.IKKE_AVSLUTTET
+        )
+        val datoerUtenfor = createTiltaksgjennomforing(
+            startDato = lastSuccessDate,
+            avslutningsstatus = Avslutningsstatus.IKKE_AVSLUTTET
+        )
+
+        test("oppdater statuser på kafka på relevante tiltaksgjennomføringer") {
+            tiltakstypeRepository.upsert(tiltakstype)
+
+            tiltaksgjennomforingRepository.upsert(startdatoInnenforMenAvsluttetStatus)
+            tiltaksgjennomforingRepository.upsert(startdatoInnenfor)
+            tiltaksgjennomforingRepository.upsert(sluttdatoInnenforMenAvbruttStatus)
+            tiltaksgjennomforingRepository.upsert(sluttdatoInnenfor)
+            tiltaksgjennomforingRepository.upsert(datoerUtenfor)
+
+            kafkaSyncService.oppdaterTiltaksgjennomforingsstatus(today, lastSuccessDate)
+
+            verify(exactly = 1) { tiltaksgjennomforingKafkaProducer.publish(startdatoInnenfor.toDto()) }
+            verify(exactly = 1) { tiltaksgjennomforingKafkaProducer.publish(sluttdatoInnenfor.toDto()) }
+            verify(exactly = 0) { tiltaksgjennomforingKafkaProducer.publish(startdatoInnenforMenAvsluttetStatus.toDto()) }
```

Comment:
Dette er kanskje en god kanditat for `verifyAll` (i stedet for 5 `verify`'s), da slipper vi også å asserte disse `exactly = 0` som bærer med seg en risiko for at vi bommer med assert'en den forventete generering av forventet payload har en logikkbrist.